### ISSUE TITLE
Native tabs super slow on Electron v4

### ISSUE THREAD

**Issue by rafeca** [atom/atom#19611]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

OSX native tabs operations super slow under Electron v4

### Steps to Reproduce

1. Build Atom from the [Electron v4](https://github.com/atom/atom/pull/19373) branch.
2. Run it and enable native tabs by clicking on the menu "View -> Show Tab Bar"
3. Create a new window and see the delay.

**Expected behavior:**

No delay when creating windows, merging windows, etc.

**Actual behavior:**

There's some delay that blocks the Atom UI for 5s-10s

**Reproduces how often:**

100%

### Versions

Atom v1.40.0 dev under Electron v4.2.5

### Additional Information

macOS version: 10.14.5 (18F132)
(only reproducible on macOS since this option is only available here).

**Comment by rafeca** [comment]
After some investigation, this seems to be caused by the calls we do to `remote.getCurrentWindow()` to get the window dimensions and store them for next session.

This call somehow got much slower in some situations on Electron v4, I've created [an issue](https://github.com/electron/electron/issues/19027) on the electron repository.

**Comment by rafeca** [comment]
Ok, more investigation done: seems like there are two separate issues:
1. The really anormal slowness when using native tabs: This was fixed by https://github.com/electron/electron/pull/18314 which got included in Electron v4.2.2. Somehow I was testing with a local build that used Electron v4.2.0 and I could see the issue. Once I upgraded to v4.2.5 the issue was gone 🎉
2. The slight slowness when resizing the Atom window. This is caused by https://github.com/electron/electron/issues/19027 and is not fixed yet, but we can workaround it by adding some throttling to the `resize` listener.

I'm gonna close this issue since it was about native tabs which is fixed
